app/models/zoom_meeting.rb

```diff
@@ -5,6 +5,10 @@ class ZoomMeeting < Meeting
 
   def self.from_meeting_details(meeting_url)
     meeting_id = meeting_url.split("/").last
+    if !meeting_id.nil?
+      raise invalid_zoom_url_error if meeting_id.include?(" ")
+      raise invalid_meeting_id_length_zoom_url_error if meeting_id.length < 10 || meeting_id.length > 11
+    end
```

Comment:
This is definitely on the right track! 

Some potential issues:

1) if Zoom's meeting urls change for whatever reason (for example ids now have 12 digits), then this would not work. In general we want to make our app flexible in how it works with external dependencies, but especially with Zoom because they have already made breaking changes to their api in the middle of our development.

2) This wouldn't handle any cases that create a BadURI error that aren't explicitly listed, for example if someone gives an id with the correct length and no spaces, but includes another invalid character like a return.

For these reasons, I think it would be best to use a rescue block to handle the bad uri error that's thrown in the ZoomService.

Reply:
What I'm thinking now is to perhaps have it filter out meeting_ids with a regex. Anything that isn't a number would get a invalid url error. 

To catch an error where the meeting id entered was too short/missing a digit. I will think more on. I think it will be more of a catching error situation. I'll let you know!

Reply:
The regex would definitely be an improvement, but we would still have the issue that we have to specify what constitutes and invalid uri. Instead, let the InvalidURI error get thrown to indicate a bad input, catch it, and handle it.

Reply:
Got it! That makes sense.

Reply:
I implemented the solution we discussed. Let me know what you think!